### PR TITLE
[prometheus-cloudwatch-exporter] Allow container security context to be configured

### PR DESCRIPTION
**Pull request by borisputerka**
Signed-off-by: Boris Puterka <boris.puterka@gmail.com>

#### What this PR does / why we need it:
It's just a small thing, but it allows us to configure securityContext on container level.
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Special notes for your reviewer:
Normally I would use `.Values.podSecurityContext` and `.Values.securityContext` but this allows backward compatibility.
#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by monotek — APPROVED** [review]

**Comment by borisputerka** [comment]
Waiting for code owners review.

**Review by monotek — APPROVED** [review]

**Comment by monotek** [comment]
@prometheus-community/helm-charts-admins please review.

**Review by torstenwalter — APPROVED** [review]

**Pull request merged**
